### DISCUSSION TITLE
Are new virtual machines for each platform a possibility?

### DISCUSSION THREAD

**Discussion by Wysardry** [davidgiven/ack#296] [Q&A]
As there doesn't seem to be many existing configuration files for specific 8-bit machines, would it be possible (and practical) to create virtual machines for use with ACK instead?

Could ACK output byte code for such machines, or is this idea just pie in the sky?

**Comment by davidgiven** [comment]
Yes. A while back I did implement most of a code generator and interpreter for a p-code system. I can't find it now, though. The architecture is most suited for a stack-based machine with each stack frame containing an array of locals. (In fact, that's pretty much what em is.)

**Comment by Wysardry** [comment]
I was thinking that a P-machine might be a good start, as Pascal is already available on the front end.

Would the other languages work with something like the P-machine automatically?

Could ACK be used to create something like a [ZIL](https://www.ifwiki.org/ZIL) to Z-machine toolchain without a major rewrite of the existing code?
